src/main/java/service/operation/BalanceOperation.java

```diff
@@ -0,0 +1,19 @@
+package service.operation;
+
+import data.db.Storage;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+
+    @Override
+    public void operation(FruitTransaction.Operation operation, FruitTransaction fruitTransaction) {
+        for (Map.Entry<String, Integer> fruit : Storage.fruitsStorage.entrySet()) {
+            if (fruit.getKey().equals(fruitTransaction.getFruit())) {
+                fruit.setValue(fruitTransaction.getQuantity());
+                return;
+            }
+        }
```

Comment:
This loop can be optimized. Instead of iterating over the map's entry set, you can directly check if the fruit is present in the storage and update its value. This will improve the performance, especially when dealing with a large number of fruits.